src/datumaro/plugins/configurable_validator.py

```diff
@@ -0,0 +1,967 @@
+        few_samples_thr=None,
+        imbalance_ratio_thr=None,
+        far_from_mean_thr=None,
+        dominance_ratio_thr=None,
+        topk_bins=None,
+    ):
+        self.tasks = tasks
+        self.warnings = warnings
+
+        self.init_flag = False
+        self.all_stats = {task: None for task in tasks}
+
+        self.few_samples_thr = few_samples_thr if few_samples_thr else self.DEFAULT_FEW_SAMPLES_THR
+        self.imbalance_ratio_thr = (
+            imbalance_ratio_thr if imbalance_ratio_thr else self.DEFAULT_IMBALANCE_RATIO_THR
+        )
+        self.far_from_mean_thr = (
+            far_from_mean_thr if far_from_mean_thr else self.DEFAULT_FAR_FROM_MEAN_THR
+        )
+        self.dominance_thr = (
+            dominance_ratio_thr if dominance_ratio_thr else self.DEFAULT_DOMINANCE_RATIO_THR
+        )
+        self.topk_bins_ratio = topk_bins if topk_bins else self.DEFAULT_TOPK_BINS
+
+    def _init_stats_collector(self, label_categories):
+        for task in self.tasks:
+            kwargs = {
+                "few_samples_thr": self.few_samples_thr,
+                "imbalance_ratio_thr": self.imbalance_ratio_thr,
+                "far_from_mean_thr": self.far_from_mean_thr,
+                "dominance_thr": self.dominance_thr,
+                "topk_bins_ratio": self.topk_bins_ratio,
+            }
+            if task == TaskType.classification:
+                self.all_stats[task] = ClsStats(
+                    label_categories=label_categories, warnings=self.warnings, **kwargs
+                )
+            elif task == TaskType.detection:
+                self.all_stats[task] = DetStats(
+                    label_categories=label_categories, warnings=self.warnings, **kwargs
+                )
+            elif task == TaskType.segmentation:
+                self.all_stats[task] = SegStats(
+                    label_categories=label_categories, warnings=self.warnings, **kwargs
+                )
+        self.init_flag = True
+
+    def _get_stats_collector(self, ann_type):
+        if TaskType.classification in self.tasks and ann_type == AnnotationType.label:
+            return self.all_stats[TaskType.classification]
+        elif TaskType.detection in self.tasks and ann_type == AnnotationType.bbox:
+            return self.all_stats[TaskType.detection]
+        elif TaskType.segmentation in self.tasks and ann_type in [
+            AnnotationType.mask,
+            AnnotationType.polygon,
+            AnnotationType.ellipse,
+        ]:
+            return self.all_stats[TaskType.segmentation]
+        else:
+            return None
+
+    def compute_statistics(self, dataset):
+        if not self.init_flag:
+            self._init_stats_collector(label_categories=dataset.categories()[AnnotationType.label])
+
+        for item in dataset:
+            for stats_collector in self.all_stats.values():
+                if not stats_collector:
+                    continue
```

Comment:
Do we need this if statement? I think that L892 and `self._init_stats_collector()` make there is no `None` in `self.all_stats` without any exception. Similar to my above comment, wouldn't it be better to let `self._init_stats_collector()` return `all_stats` rather than storing it as the member variable? The `all_stats` are dependent to `dataset`, so that maintaining it as a member variable seems not good for understanding the data flow in a function.